asteroid/data/musdb18_dataset.py

```diff
@@ -0,0 +1,174 @@
+                 seq_duration=None,
+                 samples_per_track=1,
+                 random_chunks=False,
+                 random_track_mix=False,
+                 source_augmentations=lambda audio: audio,
+                 sample_rate=44100):
+
+        """MUSDB18 torch.data.Dataset that samples from the MUSDB18 tracks
+        using track and excerpts with replacement.
+
+        Parameters
+        ----------
+        root : str
+            root path of MUSDB18HQ, please download from zenodo
+        suffix : str
+            specify the filename suffice, defaults to `.wav`
+        split : str
+            select dataset subfolder, defaults to ``train``.
+        subsets : list(str)
+            selects a list of track folders, defaults to `None` 
+            which loads all tracks
+        seq_duration : float
+            training is performed in chunks of ``seq_duration`` (in seconds,
+            defaults to ``None`` which loads the full audio track
+        samples_per_track : int
+            sets the number of samples, yielded from each track
+            Defaults to 1
+        source_augmentations : list[callables]
+            provide list of augmentation function that take a multi-channel
+            audio file of shape (src, samples) as input and output. Defaults to
+            no-augmentations (input = output)
+        random_track_mix : boolean
+            randomly mixes sources from different tracks to assemble a
+            custom mix. This augmenation is only applied for the train subset.
+        args, kwargs : additional keyword arguments
+            used to add further control for the musdb dataset
+            initialization function.
+        """
+        self.root = Path(root).expanduser()
+        self.split = split
+        self.sample_rate = sample_rate
+        self.seq_duration = seq_duration
+        self.random_track_mix = random_track_mix
+        self.random_chunks = random_chunks
+        self.source_augmentations = source_augmentations
+        self.source_files = source_files
+        self.suffix = suffix
+        self.subset = subset
+        self.samples_per_track = samples_per_track
+        self.tracks = list(self.get_tracks())
+        if not self.tracks:
+            raise RuntimeError("No tracks found")
+
+    def __getitem__(self, index):
+        # assemble the mixture of target and interferers
+        audio_sources = {}
+        # load interferers
+        for source in self.source_files:
+            # optionally select a random track for each source
+            if self.random_track_mix:
+                track_id = random.choice(range(len(self.tracks)))
+            else:
+                track_id = index // self.samples_per_track
+
+            track_path = self.tracks[track_id]['path']
+            if self.random_chunks:
+                min_duration = self.tracks[track_id]['min_duration']
+                start = random.uniform(0, min_duration - self.seq_duration)
```

Comment:
I didn't run the code but it looks like start is not defined from `random_chunks=False` (default)

Reply:
Also, the start from different instruments will be different right? Is this intended?

Reply:
And `min_duration` is None if `seq_duration` is None, but `random_chunks` can still be True.

Reply:
> I didn't run the code but it looks like start is not defined from random_chunks=False (default)

good catch

Reply:
> Also, the start from different instruments will be different right? Is this intended?

yes, it helps a lot. I might want to make this optional, indeed

Reply:
| made this behavior optional and fixed the other issues